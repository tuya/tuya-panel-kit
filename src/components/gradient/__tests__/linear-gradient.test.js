import { Text } from 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import LinearGradient from '../linear-gradient';

const linearGradientBackground = {
  '3%': '#FF7E38',
  '90%': '#FF624C',
};

describe('LinearGradient Component', () => {
  it('componentWillReceiveProps(nextProps)', () => {
    const wrapper = shallow(<LinearGradient {...linearGradientBackground} />);
    wrapper.setProps({
      stops: {
        '50%': 'rgb(255, 255, 255)',
        '100%': 'rgb(0, 0, 0)',
      },
    });
  });
  it('basic render', () => {
    const component = renderer.create(<LinearGradient {...linearGradientBackground} />);

    expect(component.root.props).toEqual({
      '3%': '#FF7E38',
      '90%': '#FF624C',
      children: null,
      gradientId: 'linear-gradient',
      stops: { '0%': 'rgb(255, 255, 255)', '100%': 'rgb(0, 0, 0)' },
      style: null,
      x1: '0%',
      x2: '0%',
      y1: '0%',
      y2: '100%',
    });
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('LinearGradient setSource', () => {
    const component = renderer.create(<LinearGradient {...linearGradientBackground} />);

    component.root.instance.setSource({
      '2%': '#FF7E38',
      '90%': '#FF624C',
    });
    expect(component.root.instance.state).toEqual({
      stops: {
        '2%': '#FF7E38',
        '90%': '#FF624C',
      },
      x1: '0%',
      x2: '0%',
      y1: '0%',
      y2: '100%',
    });
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('LinearGradient render children', () => {
    const component = renderer.create(
      <LinearGradient {...linearGradientBackground}>
        <Text>111111111111111</Text>
      </LinearGradient>
    );

    expect(component.root.instance.props.children.props.children).toEqual('111111111111111');
    expect(component.toJSON()).toMatchSnapshot();
  });
});
