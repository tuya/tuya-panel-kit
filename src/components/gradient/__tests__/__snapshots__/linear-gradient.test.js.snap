// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinearGradient Component LinearGradient render children 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "bottom": 0,
        "flex": 1,
        "height": 1334,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      },
      null,
    ]
  }
>
  <RNSVGSvgView
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        false,
        Object {
          "flex": 0,
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <RNSVGDefs>
      <RNSVGLinearGradient
        gradient={
          Array [
            1,
            1,
            1,
            1,
            0,
            0,
            0,
            1,
            0,
            1,
          ]
        }
        gradientTransform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        gradientUnits={0}
        name="linear-gradient"
        x1="0%"
        x2="0%"
        y1="0%"
        y2="100%"
      />
    </RNSVGDefs>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      fill="url(#linear-gradient)"
    >
      111111111111111
    </Text>
  </RNSVGSvgView>
</View>
`;

exports[`LinearGradient Component LinearGradient setSource 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "bottom": 0,
        "flex": 1,
        "height": 1334,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      },
      null,
    ]
  }
>
  <RNSVGSvgView
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        false,
        Object {
          "flex": 0,
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <RNSVGDefs>
      <RNSVGLinearGradient
        gradient={
          Array [
            1,
            0.49411764705882355,
            0.2196078431372549,
            1,
            1,
            0.3843137254901961,
            0.2980392156862745,
            1,
            0.02,
            0.9,
          ]
        }
        gradientTransform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        gradientUnits={0}
        name="linear-gradient"
        x1="0%"
        x2="0%"
        y1="0%"
        y2="100%"
      />
    </RNSVGDefs>
  </RNSVGSvgView>
</View>
`;

exports[`LinearGradient Component basic render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "bottom": 0,
        "flex": 1,
        "height": 1334,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      },
      null,
    ]
  }
>
  <RNSVGSvgView
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        false,
        Object {
          "flex": 0,
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <RNSVGDefs>
      <RNSVGLinearGradient
        gradient={
          Array [
            1,
            1,
            1,
            1,
            0,
            0,
            0,
            1,
            0,
            1,
          ]
        }
        gradientTransform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        gradientUnits={0}
        name="linear-gradient"
        x1="0%"
        x2="0%"
        y1="0%"
        y2="100%"
      />
    </RNSVGDefs>
  </RNSVGSvgView>
</View>
`;
